### PR TITLE
Document inversion ladder in curve25519

### PR DESCRIPTION
**Pull request by ekasper**
This demystifies two for-loops that do nothing. They were used to write
the ladder in a unified way. Now that the ladder is otherwise commented,
remove the dead loops.

### CODE REVIEW

**Comment by richsalz** [comment]
+1

File: crypto/ec/curve25519.c [hunk 1]
```diff
@@ -670,60 +670,93 @@ static void fe_invert(fe out, const fe z) {
...
+
+  /* t0 = z ** 2 */
   fe_sq(t0, z);
-  for (i = 1; i < 1; ++i) {
-    fe_sq(t0, t0);
-  }
+
+  /* t1 = t0 ** 3 = z ** 8 */
   fe_sq(t1, t0);
   for (i = 1; i < 2; ++i) {
```
Comment (mattcaswell) [inline crypto/ec/curve25519.c]: Why do we need a "for" loop here at all?

File: crypto/ec/curve25519.c [hunk 2]
```diff
@@ -670,60 +670,93 @@ static void fe_invert(fe out, const fe z) {
...
+   * 2 ** 255 - 21 = (2 ** 5) * (2 ** 250 - 1) + 11.
+   */
+
+  /* t0 = z ** 2 */
   fe_sq(t0, z);
-  for (i = 1; i < 1; ++i) {
-    fe_sq(t0, t0);
-  }
+
+  /* t1 = t0 ** 3 = z ** 8 */
```
Comment (mattcaswell) [inline crypto/ec/curve25519.c]: t0 ** 4??

Comment (ekasper) [inline crypto/ec/curve25519.c]: Good catch! Fixed.

File: crypto/ec/curve25519.c (hunk 1 above)
Comment (ekasper) [inline crypto/ec/curve25519.c]: Removed; like the dead loops, this was just for unifying the representation. I think we don't need it now that we have comments.

**Comment by mattcaswell** [comment]
+1

**Comment by mattcaswell** [comment]
@ekasper can this go in now so we can close this PR?

**Comment by ekasper** [comment]
Merged.

**Pull request closed without merging**